x-pack/plugins/observability_solution/observability_ai_assistant/server/service/task_manager_definitions/register_migrate_knowledge_base_entries_task.ts

```diff
@@ -25,50 +27,58 @@ export async function registerMigrateKnowledgeBaseEntriesTask({
   taskManager,
   logger,
   core,
+  config,
 }: {
   taskManager: TaskManagerSetupContract;
   logger: Logger;
   core: CoreSetup<ObservabilityAIAssistantPluginStartDependencies>;
+  config: ObservabilityAIAssistantConfig;
 }) {
-  logger.debug(`Register task "${TASK_TYPE}"`);
-
-  const [coreStart, pluginsStart] = await core.getStartServices();
-
-  taskManager.registerTaskDefinitions({
-    [TASK_TYPE]: {
-      title: 'Migrate AI Assistant Knowledge Base',
-      description: `Migrates AI Assistant knowledge base entries`,
-      timeout: '1h',
-      maxAttempts: 5,
-      createTaskRunner() {
-        return {
-          async run() {
-            logger.debug(`Run task: "${TASK_TYPE}"`);
-
-            const esClient = { asInternalUser: coreStart.elasticsearch.client.asInternalUser };
-            await runSemanticTextKnowledgeBaseMigration({ esClient, logger });
-          },
-        };
+  try {
+    logger.debug(`Register task "${TASK_TYPE}"`);
+
+    const [coreStart, pluginsStart] = await core.getStartServices();
+
+    taskManager.registerTaskDefinitions({
```

Comment:
possibly stupid question: shouldn't this have an interval defined?

Reply:
or if the interval is not set, it should only run once, forever?

Reply:
I'm not expert on tasks but afaiu the tasks is registered, but not scheduled here.
Then we can schedule the task with `pluginsStart.taskManager.ensureScheduled`. This takes an `interval` property. This is optional so I would expect it is only run once if omitted. The docs are not super detailed about the default behaviour:

https://github.com/elastic/kibana/blob/e9ee5b4765f89f236916c9d2952006f78d6519a7/x-pack/plugins/task_manager/server/task.ts#L292-L297

Reply:
@pmuellr can you confirm that omitting `interval` for `ensureScheduled` means it will only run once?
Also, what happens if the task is scheduled and executed via `ensureScheduled` and `runSoon` is called? Will it run the task again?

Reply:
Sorry about the timing of this reply, missed it over the US holiday.

Yes, you can omit the `schedule` and it will then only run once.